docs/proposals-accepted/202304-cardinalitymanagment.md

```diff
@@ -0,0 +1,190 @@
+---
+type: proposal
+title: Series Cardinality API
+status: accepted
+owner: jatinagwal
+menu: proposals-accepted
+---
+
+### Related links/tickets
+
+- https://github.com/thanos-io/thanos/issues/6007
+- https://mentorship.lfx.linuxfoundation.org/project/dbce5279-d029-46f3-b117-9e9dd7f84bd6
+
+### Why
+
+At present, multiple product teams send their metrics to Thanos. However, when product teams send metrics with high cardinality to Thanos, it can overwhelm the Store Gateway and Queriers, preventing any teams from running queries. To prevent this issue from occurring, we should have a cardinality management support in Thanos. This would allow teams to monitor the cardinality of both labels and metrics.
+
+### Pitfalls of current solutions
+
+Currently, Thanos receiver supports the TSDB stats API on remote write addresses, which provides information about series cardinality. However, this API can only return a limited number of results per query, typically no more than 10. This limitation makes it difficult to track arbitrary metrics and may lead to incomplete or inaccurate cardinality data. To address this issue, there should be some Top-K selector that can return the stats of the top K entries, allowing teams to more effectively monitor cardinality and identify potential issues.
+
+### Goals
+
+Implement APIs exposing series cardinalities in Thanos.
+
+### Proposal
+
+We will be Developing a new API in Thanos to calculate series cardinality through Thanos Receiver.
+
+#### API Design
+
+The new API will have the following endpoint-
+
+`GET /api/v1/cardinality`
+
+It will retrieve the series cardinality information.
+
+The API will accept the following query parameters:
+
+- topK : (optional) An integer specifying the top K results to return for each category. Default is 10.
```

Comment:
Top k means top k series? Or top k for all results? I asked this because `seriesCountByLabelValuePair` could definitely have more items than others. If we do topk for all results then `seriesCountByLabelValuePair` is incomplete actually